specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,577 @@
+  /// The Javascript will receive `MediaStreamTrack::ended` event when the API
+  /// is called.
+  /// The Javascript can restart the stream with getTextureStream.
+  /// The API call will release any internal resources on both of WebView2 host
+  /// and the browser processes.
+  /// API calls of Present, CreateBuffer will fail after this
+  /// with an error of COREWEBVIEW2_TEXTURE_STREAM_ERROR_NO_VIDEO_TRACK_STARTED.
+  /// The Stop API will be called implicitly when ICoreWebView2StagingTextureStream
+  /// object is destroyed.
+  HRESULT Stop();
+  /// Event handler for those that occur at the Renderer side, the example
+  /// are CreateBuffer, Present, or Stop.
+  HRESULT add_ErrorReceived(
+      [in] ICoreWebView2StagingTextureStreamErrorReceivedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for ErrorReceived event.
+  HRESULT remove_ErrorReceived([in] EventRegistrationToken token);
+  /// Updates d3d Device when it is updated by RenderAdapterLUIDChanged
+  /// event.
+  HRESULT SetD3DDevice([in] IUnknown* d3dDevice);
+  /// Event handler for receiving texture by Javascript.
+  /// `window.chrome.webview.registerTextureStream` call by Javascript will
+  /// request sending video frame to the host where it will filter requested
+  /// page's origin against allowed origins. If allowed, the Javascript will
+  /// send a video frame (web texture), through MediaStreamTrack insertable APIs,
+  /// MediaStreamTrackGenerator.
+  /// https://www.w3.org/TR/mediacapture-transform/.
+  /// WebTextureReceived event will be called only when it receives
+  /// a web texture. There is no start event for receiving web texture.
+  HRESULT add_WebTextureReceived(
+      [in] ICoreWebView2StagingTextureStreamWebTextureReceivedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for WebTextureReceived event.
+  HRESULT remove_WebTextureReceived([in] EventRegistrationToken token);
+  /// Event handler for stopping of the receiving texture stream.
+  /// It is expected that the host releases any holding handle/resource from
+  /// the WebTexture before an event handler returns.
+  HRESULT add_WebTextureStreamStopped(
+      [in] ICoreWebView2StagingTextureStreamWebTextureStreamStoppedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for WebTextureStreamStopped event.
+  HRESULT remove_WebTextureStreamStopped([in] EventRegistrationToken token);
+}
+/// TextureBuffer that the host writes to so that the Renderer
+/// will render on it.
+[uuid(0836f09c-34bd-47bf-914a-99fb56ae2d07), object, pointer_default(unique)]
+interface ICoreWebView2StagingTexture : IUnknown {
+    /// Returns shared Windows NT handle. The caller expected to open it with
+    /// ID3D11Device1::OpenSharedResource1 and writes the incoming texture to it.
+    [propget] HRESULT Handle([out, retval] HANDLE* handle);

```

Comment:
Update documentation to say that the caller should not close the handle. The texture object owns closing it. The underlying shared memory may be reused by WebView2.